sql: TestTrace failed

sql.TestTrace [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20706939?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20706939?buildTab=artifacts#/) on master @ [c9eda2b92a04af26604cffb20f18ece56e0fd62b](https://github.com/cockroachdb/cockroach/commits/c9eda2b92a04af26604cffb20f18ece56e0fd62b):


```
        looked up range descriptor: et:desc:r84:/Tenant/10/Table/1{2-07/1/5} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=9], lease:<empty>
        key: /Tenant/10/Table/106/1, desc: r84:/Tenant/10/Table/1{2-07/1/5} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=9]
        routing to nearest replica; leaseholder not required order=[(n1,s1):1(health=false match=2 latency=0µs),(n2,s2):3(health=false match=1 latency=0µs),(n3,s3):2(health=false match=1 latency=0µs)]
        r84: sending batch 1 Scan to (n1,s1):1
        failed to perform look up for node descriptor node descriptor with node ID 0 was not found
        sending request to 127.0.0.1:40089
        sending batch request
        === SPAN START: /cockroach.roachpb.Internal/Batch ===
        === SPAN START: /cockroach.roachpb.Internal/Batch ===
        node received request: 1 Scan
        executing Scan [/Tenant/10/Table/106/1,/Tenant/10/Table/106/2), [txn: f0045f48], [can-forward-ts], [max_span_request_keys: 0], [target_bytes: 58701]
        read-only path
        sequencing request
        acquiring latches
        scanning lock table for conflicting locks
        replica.Send got error: key range /Tenant/10/Table/106/1-/Tenant/10/Table/106/2 outside of bounds of range /Tenant/10/Table/12-/Tenant/10/Table/25; suggested ranges: [desc: r84:/Tenant/10/Table/{12-25} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=10], lease: repl=(n1,s1):1 seq=1 start=0,0 type=LeaseExpiration exp=1762467366.927588724,0 pro=1762467360.927588724,0 acq=Request, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING]
        read-only path
        sequencing request
        acquiring latches
        scanning lock table for conflicting locks
        can drop latches early for batch (Scan [/Tenant/10/Table/106/1,/Tenant/10/Table/106/2), [txn: f0045f48], [can-forward-ts], [max_span_request_keys: 0], [target_bytes: 58701]); scanning lock table first to detect conflicts
        lock table scan stats: seeked 1 times (1 internal); stepped 0 times (0 internal)
        lock table scan complete without conflicts; dropping latches early
        executing read-only batch
        evaluated Scan command header:<key:"\376\222\362\211" end_key:"\376\222\362\212" > scan_format:BATCH_RESPONSE , txn="sql txn" meta={id=f0045f48 key=/Min iso=Serializable pri=0.07246407 epo=0 ts=1762467364.020124603,0 min=1762467364.020124603,0 seq=0} lock=false stat=PENDING rts=1762467364.020124603,0 gul=1762467364.520124603,0 obs={n1@1762467364.022732800,0} : resp=header:<> col_batches:<> , err=<nil>
        n1 scan stats: stepped 0 times (0 internal); seeked 1 times (1 internal); block-bytes: (total 0 B, cached 0 B, duration 0s); points: (count 0, key-bytes 0 B, value-bytes 0 B, tombstoned: 0) ranges: (count 0), (contained-points 0, skipped-points 0) evaluated requests: 0 gets, 1 scans, 0 reverse scans
        read completed
        node sending response
        received batch response
        received updated range info: [desc: r84:/Tenant/10/Table/{12-25} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=10], lease: repl=(n1,s1):1 seq=1 start=0,0 type=LeaseExpiration exp=1762467366.927588724,0 pro=1762467360.927588724,0 acq=Request, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING desc: r85:/Tenant/10/Table/{25-107/1/5} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=10], lease: repl=(n1,s1):1 seq=1 start=0,0 type=LeaseExpiration exp=1762467366.927588724,0 pro=1762467360.927588724,0 acq=Request, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING]
        evict cached descriptor: desc=desc:r84:/Tenant/10/Table/1{2-07/1/5} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=9], lease:<empty>
        evicting cached range descriptor with 2 replacements
        clearing entries overlapping r84:/Tenant/10/Table/{12-25} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=10]
        clearing entries overlapping r85:/Tenant/10/Table/{25-107/1/5} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=10]
        r_u:0.625 
        recording span to refresh: /Tenant/10/Table/106/{1-2}
        ComponentStats{ID: {0949e576-adfa-4809-a025-d99b9ae033a3 PROCESSOR 0 1 }, KV nodes: n1, KV regions: test, KV time: 3ms, KV rows decoded: 0, KV bytes read: 0 B, KV gRPC calls: 1, KV pairs read: 0, MVCC step count (ext/int): 0/0, MVCC seek count (ext/int): 1/1, rows output: 0}
        ComponentStats{ID: {0949e576-adfa-4809-a025-d99b9ae033a3 FLOW 0 1 test}}
        execution ends
        rows affected: 0
        === SPAN START: commit sql txn ===
        AutoCommit. err: <nil>
        releasing 4 descriptors
        release: 100("defaultdb",0101804e309461672e4691bd1f16730a04b7bc) ver=1:0,0, refcount=0
        release: 104("test",0101804e309461672e4691bd1f16730a04b7bc) ver=1:0,0, refcount=0
        release: 105("public",0101804e309461672e4691bd1f16730a04b7bc) ver=1:0,0, refcount=0
        release: 106("foo",0101804e309461672e4691bd1f16730a04b7bc) ver=1:0,0, refcount=0
        [NoTxn pos:10] executing ExecStmt: SET TRACING = off
        executing: SET TRACING = off in state: NoTxn
--- FAIL: TestTrace/Session/TracingOff/node-0 (0.17s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTrace.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-56355

--- a/pkg/sql/trace_test.go
+++ b/pkg/sql/trace_test.go
@@ -37,10 +37,10 @@ import (
 
 func TestTrace(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
 	skip.UnderRace(t, "does too much work under race, see: "+
 		"https://github.com/cockroachdb/cockroach/pull/56343#issuecomment-733577377")
-
-	defer log.Scope(t).Close(t)
 
 	// These are always appended, even without the test specifying it.
 	alwaysOptionalSpans := []string{
@@ -61,33 +61,30 @@ func TestTrace(t *testing.T) {
 		"/cockroach.sql.distsqlrun.DistSQL/FlowStream",
 		"noop",
 	}
-	nonVectorizedExpSpans := []string{
+	// These are optional if we use a test tenant.
+	tenantOptionalSpans := []string{
+		"/cockroach.roachpb.Internal/RangeLookup",
+		"executeWriteBatch",
+	}
+	commonExpSpans := []string{
 		"session recording",
 		"sql txn",
 		"sql query",
 		"optimizer",
 		"flow",
-		"table reader",
 		"consuming rows",
 		"txn coordinator send",
 		"dist sender send",
 		"/cockroach.roachpb.Internal/Batch",
 		"commit sql txn",
 	}
-	vectorizedExpSpans := []string{
-		"session recording",
-		"sql txn",
-		"sql query",
-		"optimizer",
-		"flow",
+	nonVectorizedExpSpans := append(append([]string(nil), commonExpSpans...), []string{
+		"table reader",
+	}...)
+	vectorizedExpSpans := append(append([]string(nil), commonExpSpans...), []string{
 		"batch flow coordinator",
 		"colbatchscan",
-		"consuming rows",
-		"txn coordinator send",
-		"dist sender send",
-		"/cockroach.roachpb.Internal/Batch",
-		"commit sql txn",
-	}
+	}...)
 
 	getRows := func(t *testing.T, sqlDB *gosql.DB, distsql, vectorize string, useShowTraceFor bool) (*gosql.Rows, string, error) {
 		if _, err := sqlDB.Exec(fmt.Sprintf("SET distsql = %s", distsql)); err != nil {
@@ -210,6 +207,9 @@ func TestTrace(t *testing.T) {
 		optionalSpans := append([]string{}, alwaysOptionalSpans...)
 		if test.distSQL == "on" {
 			optionalSpans = append(optionalSpans, distsqlOptionalSpans...)
+		}
+		if cluster.StartedDefaultTestTenant() {
+			optionalSpans = append(optionalSpans, tenantOptionalSpans...)
 		}
 		expSpans := nonVectorizedExpSpans
 		if test.vectorize == "on" {

